test/grpc_test.go

```diff
@@ -56,3 +59,77 @@ func TestGrpc(t *testing.T) {
 		t.Errorf("Expected 2 RRs in additional section, but got %d", len(d.Extra))
 	}
 }
+
+type grpcDnsServiceServer struct {
+	pb.UnimplementedDnsServiceServer
+}
+
+func (*grpcDnsServiceServer) Query(ctx context.Context, in *pb.DnsPacket) (*pb.DnsPacket, error) {
+	msg := &dns.Msg{}
+	msg.Unpack(in.GetMsg())
+	answer := new(dns.Msg)
+	answer.SetRcode(msg, dns.RcodeSuccess)
+	buf, _ := answer.Pack()
+	return &pb.DnsPacket{Msg: buf}, nil
+}
+
+func TestGrpcUnix(t *testing.T) {
+	fd, err := os.MkdirTemp("", "*.grpc")
```

Comment:
`MkdirTemp` creates the directory. It uses an empty pattern to create only the directory and uses `filepath.Join(d, "sock")` to create the file.